Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -361,6 +361,29 @@ struct Gx2fSystem {
   std::size_t m_ndf = 0u;
 };
 
+/// @brief Adds a measurement to the GX2F equation system in a modular backend function.
+///
+/// This function processes measurement data and integrates it into the GX2F
+/// system.
+///
+/// @param extendedSystem All parameters of the current equation system to update.
+/// @param jacobianFromStart The Jacobian matrix from the start to the current state.
+/// @param covarianceMeasurement The covariance matrix of the measurement.
+/// @param predicted The predicted state vector based on the track state.
+/// @param measurement The measurement vector.
+/// @param projector The projection matrix.
+/// @param logger A logger instance.
+///
+/// @note The dynamic Eigen matrices are suboptimal. We could think of
+/// templating again in the future on kMeasDims. We currently use dynamic
+/// matrices to reduce the memory during compile time.
+void addMeasurementToGx2fSumsBackend(
```

Comment:
Can this just be a private function in the GX2F or does this need to be public?

Reply:
I think, we could move all the functions into `class Gx2Fitter`. How about a `detail` namespace?

I am a bit reluctant to add more to the `class Gx2Fitter` since, it is already quite large. But I would be interested in pushing all the stuff into a new header like gx2f-helpers.

I could do a further refactor after this PR?